webapp/src/test/java/com/box/l10n/mojito/service/ai/translation/AITranslateJobTest.java

```diff
@@ -0,0 +1,314 @@
+            eq(TMTextUnitVariant.Status.MT_TRANSLATED),
+            eq(false),
+            isA(ZonedDateTime.class));
+    verify(tmService, times(1))
+        .addTMTextUnitVariant(
+            eq(1L),
+            eq(4L),
+            eq("translated"),
+            eq("comment"),
+            eq(TMTextUnitVariant.Status.MT_TRANSLATED),
+            eq(false),
+            isA(ZonedDateTime.class));
+    verify(tmTextUnitPendingMTRepository, times(1)).delete(isA(TmTextUnitPendingMT.class));
+  }
+
+  @Test
+  public void testTranslateFailure() throws Exception {
+    aiTranslateJob.call(aiTranslateJobInput);
+    when(llmService.translate(
+            isA(TMTextUnit.class), isA(String.class), isA(String.class), isA(AIPrompt.class)))
+        .thenThrow(new RuntimeException("test"));
+    verify(llmService, times(3))
+        .translate(
+            isA(TMTextUnit.class), isA(String.class), isA(String.class), isA(AIPrompt.class));
+    verify(tmTextUnitPendingMTRepository, times(1)).delete(isA(TmTextUnitPendingMT.class));
+  }
+
+  @Test
+  public void testTranslateReuseSource() throws Exception {
+    aiTranslateJob.isReuseSourceOnLanguageMatch = true;
+    aiTranslateJob.call(aiTranslateJobInput);
+    verify(llmService, times(2))
+        .translate(
+            isA(TMTextUnit.class), isA(String.class), isA(String.class), isA(AIPrompt.class));
+    verify(tmService, times(1))
+        .addTMTextUnitVariant(
+            eq(1L),
+            eq(2L),
+            eq("translated"),
+            eq("comment"),
+            eq(TMTextUnitVariant.Status.MT_TRANSLATED),
+            eq(false),
+            isA(ZonedDateTime.class));
+    verify(tmService, times(1))
+        .addTMTextUnitVariant(
+            eq(1L),
+            eq(3L),
+            eq("translated"),
+            eq("comment"),
+            eq(TMTextUnitVariant.Status.MT_TRANSLATED),
+            eq(false),
+            isA(ZonedDateTime.class));
+    verify(tmService, times(1))
+        .addTMTextUnitVariant(
+            eq(1L),
+            eq(4L),
+            eq("content"),
+            eq("comment"),
+            eq(TMTextUnitVariant.Status.MT_TRANSLATED),
+            eq(false),
+            isA(ZonedDateTime.class));
+    verify(tmTextUnitPendingMTRepository, times(1)).delete(isA(TmTextUnitPendingMT.class));
+    verify(meterRegistry, times(1))
+        .counter(eq("AITranslateJob.translate.reuseSourceAsTranslation"), any(Tags.class));
+  }
+
+  @Test
+  public void testPendingMTEntityIsExpired() throws Exception {
+    TmTextUnitPendingMT expiredPendingMT = new TmTextUnitPendingMT();
+    expiredPendingMT.setCreatedDate(ZonedDateTime.now().minusHours(4));
+    when(tmTextUnitPendingMTRepository.findByTmTextUnitId(1L)).thenReturn(expiredPendingMT);
+    aiTranslateJob.call(aiTranslateJobInput);
+    verify(llmService, times(0))
+        .translate(
+            isA(TMTextUnit.class), isA(String.class), isA(String.class), isA(AIPrompt.class));
```

Comment:
I'm not familiar with the testing framework just yet. I find it curious that you need to be explicit about the parameters of the call if you are testing that the method is not called at all

Reply:
I do see that there is a `never` option too
`verify(yourMock, never()).someMethod();`

Reply:
You have an option of `any()` as a type. 

You could have an overloaded method in java so providing the argument parameters can allow you to differentiate that the function with these types wasn't called vs another method signature with same number of params but different types.

Happy to update `times(0)` to `never()`